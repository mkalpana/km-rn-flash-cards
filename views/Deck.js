import React, { Component } from 'react';
import { View, Text, StyleSheet } from 'react-native';
import PropTypes from 'prop-types';
import { fetchDeck } from '../actions';
import { connect } from 'react-redux';
import { grey, black, white, transparent } from '../utils/colors';
import { Button } from '../components';
import { get } from 'lodash';

class Deck extends Component{
  static navigationOptions = ({ navigation }) => {
    const { title } = navigation.state.params
    return {
      title: `${title}`
    }
  };

  componentDidMount() {
    const { navigation } = this.props;
    const title = get(navigation, 'state.params.title', '');
    if (title) {
      this.props.fetchDeck(title);
    }
  }

  render() {
    const { deck, navigation } = this.props;
    return (deck && deck.title && deck.questions &&
      <View style={styles.container}>
        <View style={styles.textContainer}>
          <Text style={styles.deckTitle}>{deck.title}</Text>
          <Text style={styles.deckQuestionCount}>{deck.questions && deck.questions.length} cards</Text>
        </View>
        <View style={styles.inputContainer}>
          <Button
            style={styles.btnStyle}
            label="Add Card"
            buttonStyle={{ backgroundColor: transparent, borderWidth: 2, borderColor: black }}
            buttonTextStyle={{ color: black }}
            onPress={() => navigation.navigate('AddCard', { title: deck.title })}
          />
          { deck.questions.length > 0 && (
            <Button
              style={styles.btnStyle}
              label="Start Quiz"
              onPress={() => navigation.navigate('Quiz', { title: deck.title })}
            />
          )}
        </View>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'column',
    justifyContent: 'space-around',
    alignItems: 'stretch',
  },
  textContainer: {
    flex: 1,
    justifyContent: 'center',
    alignSelf: 'center'
  },
  inputContainer: {
    flex: 1,
    justifyContent: 'center'
  },
  deckTitle: {
    padding: 10,
    fontSize: 40,
    color: black,
    alignSelf: 'center'
  },
  deckQuestionCount: {
    padding: 10,
    fontSize: 20,
    color: grey,
    alignSelf: 'center'
  },
  btnStyle: {
    marginTop: 10,
    marginBottom: 10,
    marginLeft: 40,
    marginRight: 40
  }
});

Deck.propTypes = {
  deck: PropTypes.object.isRequired,
  fetchDeck: PropTypes.func.isRequired,
};

Deck.defaultPropTypes = {
  deck: {}
};

const mapStateToProps = (state, ownProps) => {
  const { navigation } = ownProps;
  const title = get(navigation, 'state.params.title', '');
  const decksList = state.decks;
  return {
    deck: title ? decksList[title] : {}
  };
};

export default connect(mapStateToProps, {
  fetchDeck
})(Deck);