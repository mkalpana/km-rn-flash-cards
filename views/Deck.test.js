import React from 'react';
import { Deck } from './Deck';
import renderer from 'react-test-renderer';

const deck = { title: 'React', questions: [] };

describe('Deck', () => {
  it('renders without crashing', () => {
    const rendered = renderer.create(<Deck
      deck={deck}
      fetchDeck={() => deck}
    />).toJSON();
    expect(rendered).toMatchSnapshot();
  });
});
