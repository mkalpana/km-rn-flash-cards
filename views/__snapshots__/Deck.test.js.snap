// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Deck renders without crashing 1`] = `
<View
  style={
    Object {
      "alignItems": "stretch",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "space-around",
    }
  }
>
  <View
    style={
      Object {
        "alignSelf": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "alignSelf": "center",
          "color": "black",
          "fontSize": 40,
          "padding": 10,
        }
      }
    >
      React
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "alignSelf": "center",
          "color": "grey",
          "fontSize": 20,
          "padding": 10,
        }
      }
    >
      0
       cards
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "marginBottom": 10,
          "marginLeft": 40,
          "marginRight": 40,
          "marginTop": 10,
        }
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "black",
                "borderColor": "black",
                "borderRadius": 5,
                "borderWidth": 2,
                "padding": 20,
                "paddingLeft": 40,
                "paddingRight": 40,
              },
              Object {
                "backgroundColor": "transparent",
                "borderColor": "black",
                "borderWidth": 2,
              },
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "white",
                  "fontSize": 20,
                  "textAlign": "center",
                },
                Object {
                  "color": "black",
                },
              ]
            }
          >
            Add Card
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
